test/index-intervals.spec.js

```diff
@@ -1,100 +1,142 @@
+    interval: MS_IN_DAY
+});
+
+function validate_indices(indices, interval) {
+    function _regex_for_interval(interval) {
+        switch (interval) {
+            case 'day':
+                return /day[0-9][0-9][0-9][0-9]\.[0-9][0-9]\.[0-9][0-9]/;
+            case 'week':
+                return /week[0-9][0-9][0-9][0-9]\.[0-9][0-9]?/;
+            case 'month':
+                return /month[0-9][0-9][0-9][0-9]\.[0-9][0-9]/;
+            case 'year':
+                return /year[0-9][0-9][0-9][0-9]/;
+            case 'none':
+                return /none/;
+            default:
+                throw new Error(`invalid interval ${interval}`);
+        }
+    }
+
+    var regex = _regex_for_interval(interval);
+    indices.forEach(function(index) {
+        expect(regex.test(index)).equal(true);
     });
+}
 
-    function check(interval, suffix) {
-        var index = suffix ? interval + '*' : 'none';
-        return test_utils.write(points, {index: index, indexInterval: interval})
-            .then(function(result) {
-                expect(result.errors).deep.equal([]);
-                return test_utils.verify_import(points, 'local', index);
+modes.forEach(function(mode) {
+    describe('index intervals -- ' + mode, function() {
+        afterEach(function() {
+            var id = test_utils.test_id;
+            return test_utils.clear_data(mode, `${id}*`);
+        });
+
+        function check(interval) {
+            var suffix = interval !== 'none' ? interval + '*' : interval;
+            var index = test_utils.test_id + suffix;
+            return Promise.map(points, function(pt, n) {
+                return test_utils.write([pt], {index: index, indexInterval: interval, id: mode})
+                    .then(function(result) {
+                        expect(result.errors).deep.equal([]);
+                    });
+            }, {concurrency: 5})
+            .then(function(results) {
+                return test_utils.verify_import(points, mode, index);
             })
             .then(function() {
                 return test_utils.list_indices();
             })
             .then(function(indices) {
-                var week_indices = indices.filter(function(index) {
-                    return index.substring(index.length - suffix.length) === suffix;
-                });
-
-                expect(week_indices.length).at.least(1);
-                var start = '2014-09-17T14:13:42.000Z';
-                var end = '2014-10-17T14:13:42.000Z';
-                return test_utils.read({from: start, to: end, index: index, indexInterval: interval});
+                validate_indices(indices, interval);
+                var from = new Date(points[0].time).toISOString();
+                var to_ms = new Date(_.last(points).time).getTime();
+                var to = new Date(to_ms + 1).toISOString();
+                return test_utils.read({id: mode, from: from, to: to, index: index, indexInterval: interval});
```

Comment:
`index` is the index pattern, so we read from multiple intervals here. They all have data, so c) is what this tests. I added tests for a) and b).